packages/wxt/src/core/resolve-config.ts

```diff
@@ -96,7 +96,22 @@ export async function resolveConfig(
   const publicDir = path.resolve(srcDir, mergedConfig.publicDir ?? 'public');
   const typesDir = path.resolve(wxtDir, 'types');
   const outBaseDir = path.resolve(root, mergedConfig.outDir ?? '.output');
-  const outDir = path.resolve(outBaseDir, `${browser}-mv${manifestVersion}`);
+  let outDirTemplate =
+    mergedConfig.outDirTemplate ?? `${browser}-mv${manifestVersion}`;
+  //* Resolve all variables in the template
+  outDirTemplate = outDirTemplate.replaceAll('{{browser}}', browser);
+  outDirTemplate = outDirTemplate.replaceAll(
+    '{{manifestVersion}}',
+    manifestVersion.toString(),
+  );
+  outDirTemplate = outDirTemplate.replaceAll(
+    '{{modeSuffix}}',
+    `${mode === 'production' ? '' : mode === 'development' ? '-dev' : '-${mode}'}`,
+  );
+  outDirTemplate = outDirTemplate.replaceAll('{{mode}}', mode);
+  outDirTemplate = outDirTemplate.replaceAll('{{command}}', command);
```

Comment:
This can be cleaned up a little bit.

```suggestion
  const modeSuffixes: Record<string, string | undefined> = {
    production: "",
    development: "-dev",
  };
  const outDirTemplate = (mergedConfig.outDirTemplate ?? `${browser}-mv${manifestVersion}`)
    // Resolve all variables in the template
    .replaceAll('{{browser}}', browser)
    .replaceAll('{{manifestVersion}}', manifestVersion.toString())
    .replaceAll('{{modeSuffix}}', `-${modeSuffixes[mode] ?? mode}`)
    .replaceAll('{{mode}}', mode)
    .replaceAll('{{command}}', command);
```

Reply:
Yeah idk how you would've wanted it since some hate chaining